pylutron_caseta/color_value.py

```diff
@@ -0,0 +1,192 @@
+# protocol which defines a method that returns a dictionary of the leap command given a zone string
+from abc import ABC, abstractmethod
+import colorsys
+from typing import Optional
+
+
+class ColorMode(ABC):
+    """
+    A protocol for setting the color property of a support light.
+   """
+
+    @abstractmethod
+    def get_spectrum_tuning_level_parameters(self) -> dict:
+        """
+        Gets the relevant parameter dictionary for the spectrum tuning level of the child class.
+
+        :return: spectrum tuning level parameter dictionary
+        """
+        pass
+
+    def get_white_tuning_level_parameters(self) -> dict:
+        """
+        Gets the relevant parameter dictionary for the white tuning level of the child class.
+
+        :return: white tuning level parameter dictionary
+        """
+        pass
+
+
+    @staticmethod
+    def get_color_from_leap(zone_status: dict) -> Optional["ColorMode"]:
+        """
+        Gets the color value from the leap command.
+
+        :param zone_status: leap zone status dictionary
+        :return: color value
+        """
+        if zone_status is None:
+            return None
+
+        color_status = zone_status.get("ColorTuningStatus")
+        if color_status is None:
+            return None
+
+        if "WhiteTuningLevel" in color_status:
+            kelvin = color_status["WhiteTuningLevel"]["Kelvin"]
+            return WarmCoolColorValue(kelvin)
+        elif "HSVTuningLevel" in color_status:
+            hue = color_status["HSVTuningLevel"]["Hue"]
+            saturation = color_status["HSVTuningLevel"]["Saturation"]
+            return FullColorValue(HueSaturationColorParameter(hue, saturation))
+
+        return None
+
+
+class FullColorParameter(ABC):
+    @abstractmethod
+    def get_hs(self) -> (int, int):
+        pass
```

Comment:
This class doesn't seem like it really does anything. It's just used as a parameter to FullColorValue, which immediately converts it into something else.

It seems like the point of this code is to allow setting colors using RGB, but it doesn't seem like LEAP supports that, and the emulation here doesn't work properly because it discards the value component. I don't think pylutron-caseta should emulate support for RGB if the hardware only supports HSB. This is a library for interfacing with the hardware so it should expose what the hardware supports.